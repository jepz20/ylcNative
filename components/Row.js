// @flow
import * as React from 'react'
import { View, StyleSheet } from 'react-native'
import { renderEl } from '../utils'

type Props = {
  children: React.Node
}

const Row: React.StatelessFunctionalComponent<Props> = ({ children }) => {
  return <View style={[styles.container]}>{renderEl(children)}</View>
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row'
  }
})

export { Row }
