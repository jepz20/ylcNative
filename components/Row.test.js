import React from 'react'
import { Text } from 'react-native'
import renderer from 'react-test-renderer'

import { Row } from './Row'

describe('Row', () => {
  test('renders with no props ', () => {
    const row = renderer.create(<Row />).toJSON()
    expect(row).toMatchSnapshot()
  })

  test('renders with a children as text', () => {
    const row = renderer.create(<Row value={20}>This is a test</Row>).toJSON()
    expect(row).toMatchSnapshot()
  })

  test('renders with a children as a component', () => {
    const row = renderer
      .create(
        <Row value={20}>
          <Text>Test</Text>
        </Row>
      )
      .toJSON()
    expect(row).toMatchSnapshot()
  })
})
