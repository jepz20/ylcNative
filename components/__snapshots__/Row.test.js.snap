// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row renders with a children as a component 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Test
  </Text>
</View>
`;

exports[`Row renders with a children as text 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    This is a test
  </Text>
</View>
`;

exports[`Row renders with no props  1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
      },
    ]
  }
/>
`;
